packages/x-date-pickers/src/TimeClock/Clock.tsx

```diff
@@ -45,16 +46,29 @@ export interface ClockProps extends ReturnType<typeof useMeridiemMode>, FormProp
   classes?: Partial<ClockClasses>;
 }
 
-const useUtilityClasses = (ownerState: ClockProps) => {
-  const { classes, meridiemMode } = ownerState;
+interface ClockOwnerState extends PickerOwnerState {
+  /**
+   * `true` if the clock is disabled, `false` otherwise.
+   */
+  isClockDisabled: boolean;
+  /**
+   * The current meridiem mode of the clock.
+   */
+  clockMeridiemMode: Meridiem | null;
```

Comment:
Same, I'm correctly scoping `isClockDisabled` to the clock to distinguish it from `isPickerDisabled`